### ISSUE TITLE
GUNNER dies on KS10 with ILOPR 

### ISSUE THREAD

**Issue by eswenson1** [PDP-10/its#2250]
GUNNER was written for a KA10. It uses I/O instructions (e.g. CONSZ) that are not available on a KS10.  These references are in the NCPUP logic.  A temporary fix is to comment out the invocation of this routine in the dispatch table that tells GUNNER which routines to run.  A more permanent fix would be to update this code to use more appropriate I/O instructions when assembled for a KS10.

**Comment by larsbrinkhoff** [comment]
Moreover, the 424 "FI" device was only used with DM and is not yet supported by the KA10 emulator.

**Comment by larsbrinkhoff** [comment]
I'm reading the GUNNER code to try to better understand the underlying intent.  The NCPUP checks the state of IMPUP in ITS; this is to see if the network (e.g. connection to Arpanet) is up or down.  It remembers the state to it can declare whether the network went down, or back up.  If it's down, it calls NETDWN to poke the IMP interface device.  This is where the IOT instructions come into play.  `CONO 424,22` sets the two PI channels and if they read back non-zero, the interface is considered alive.  If that didn't work, a new attempt is made after an hour has passed.  Presumably, we could try to implement an equivalent logic for the "KA" IMP.

CC original author @taa01776

**Comment by eswenson1** [comment]
Do we even have NCP enabled in our builds yet?

**Comment by larsbrinkhoff** [comment]
No, but this applies to the IMP unirregardless which networking protocol it's using.

**Comment by eswenson1** [comment]
Ah, so NCPxxx is a misnomer for these routines?

**Comment by larsbrinkhoff** [comment]
The were appropriately named when written, and then Arpanet changed.  Think of it as a charming reminder of historical context.
